subsys/usb/device/class/cdc_acm.c

```diff
@@ -288,10 +287,15 @@ static void tx_work_handler(struct k_work *work)
 
 	LOG_DBG("Got %zd bytes from ringbuffer send to ep %x", len, ep);
 
-	usb_transfer(ep, data, len, USB_TRANS_WRITE,
-		     cdc_acm_write_cb, dev_data);
+	ret = usb_transfer(ep, data, len, USB_TRANS_WRITE,
+			   cdc_acm_write_cb, dev_data);
+	if (ret != 0) {
+		(void)ring_buf_get_finish(dev_data->tx_ringbuf, 0);
+		return;
+	}
```

Comment:
If usb_transfer() fails, it disables TX interrupts until bus reset or similar. Instead, error branch should reschedule tx_work with some delay, maybe 1ms.